modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -2717,14 +2742,23 @@ bool InputMediaStream_FFMPEG::read(unsigned char** data, int* size, int* endOfFi
             continue;
         }
 
+        result = true;
         break;
     }
 
-    *data = pkt_.data;
-    *size = pkt_.size;
-    *endOfFile = false;
+#if USE_AV_INTERRUPT_CALLBACK
```

Comment:
Sorry, I'm not sure that I understand this question.
Could you describe problem with more details?

Behaviour of this function is changed in case of this [`break` (line 2723)](https://github.com/Itseez/opencv/pull/6358/files/7d9a4805978394176eefbc4e53821fc862ed39cf#diff-d8eb3db25b0d9d1bb7b4355d5d56e006R2723) - we don't return "true" in case of reached timeout. And this [`break` (line 2736)](https://github.com/Itseez/opencv/pull/6358/files/7d9a4805978394176eefbc4e53821fc862ed39cf#diff-d8eb3db25b0d9d1bb7b4355d5d56e006R2736) gets timeout deactivation. Other code paths should be fine.